crates/holo_hash/src/hash.rs

```diff
@@ -116,25 +116,37 @@ where
 impl<T: HashType> HoloHash<T> {
     /// Raw constructor: Create a HoloHash from 39 bytes, using the prefix
     /// bytes to determine the hash_type
-    pub fn from_raw_39(hash: Vec<u8>) -> HoloHashResult<Self> {
-        assert_length!(HOLO_HASH_FULL_LEN, &hash);
+    pub fn try_from_raw_39(hash: Vec<u8>) -> HoloHashResult<Self> {
+        if hash.len() != HOLO_HASH_FULL_LEN {
+            return Err(HoloHashError::BadSize);
+        }
         let hash_type = T::try_from_prefix(&hash[0..3])?;
         Ok(Self { hash, hash_type })
     }
+
     /// Raw constructor: Create a HoloHash from 39 bytes, using the prefix
-    /// bytes to determine the hash_type. Panics if hash_type does not match.
-    pub fn from_raw_39_panicky(hash: Vec<u8>) -> Self {
-        Self::from_raw_39(hash).expect("the specified hash_type does not match the prefix bytes")
+    /// bytes to determine the hash_type.
+    /// Panics if hash_type does not match or hash is incorrect length.
+    pub fn from_raw_39(hash: Vec<u8>) -> Self {
```

Comment:
This pattern with `from_xxx` and `try_from_xxx` suggests to me that in some way there is an infallible version of this operation and one where errors can happen. I'd prefer to just break the API here and have `from_xxx` return an error. I don't see why we'd ever want to panic when doing this, especially as we get these hashes as external inputs

Reply:
Yep please, that sounds good to me!